{"name": "tensorflow", "cve_id": "CVE-2022-23563", "publish_date": "2022-02-04T23:15Z", "cwe_id": "CWE-668", "owner": "Google Inc.", "affected_versions": [{"version_value": "2.8.0","operator": "<"}], "references": ["https://github.com/tensorflow/tensorflow/security/advisories/GHSA-wc4g-r73w-x8mm", "https://nvd.nist.gov/vuln/detail/CVE-2022-23563", "https://github.com/advisories/GHSA-wc4g-r73w-x8mm"], "description": "From NVD:\nTensorflow is an Open Source Machine Learning Framework. In multiple places, TensorFlow uses `tempfile.mktemp` to create temporary files. While this is acceptable in testing, in utilities and libraries it is dangerous as a different process can create the file between the check for the filename in `mktemp` and the actual creation of the file by a subsequent operation (a TOC/TOU type of weakness). In several instances, TensorFlow was supposed to actually create a temporary directory instead of a file. This logic bug is hidden away by the `mktemp` function usage. We have patched the issue in several commits, replacing `mktemp` with the safer `mkstemp`/`mkdtemp` functions, according to the usage pattern. Users are advised to upgrade as soon as possible.\nFrom Github:\n### Impact\nIn multiple places, TensorFlow uses `tempfile.mktemp` to create temporary files. While this is acceptable in testing, in utilities and libraries it is dangerous as a different process can create the file between the check for the filename in `mktemp` and the actual creation of the file by a subsequent operation (a TOC/TOU type of weakness).\n\nIn several instances, TensorFlow was supposed to actually create a temporary directory instead of a file. This logic bug is hidden away by the `mktemp` function usage.\n\n### Patches\nWe have patched the issue in several commits, replacing `mktemp` with the safer `mkstemp`/`mkdtemp` functions, according to the usage pattern.\nThe fix will be included in TensorFlow 2.8.0. We will also cherrypick this commit on TensorFlow 2.7.1, TensorFlow 2.6.3, and TensorFlow 2.5.3, as these are also affected and still in supported range.\n\n### For more information\nPlease consult [our security guide](https://github.com/tensorflow/tensorflow/blob/master/SECURITY.md) for more information regarding the security model and how to contact us with issues and questions.\n\n### Attribution\nThis vulnerability has been reported on huntr.dev for one scenario and discovered via variant analysis on other instances.", "latest_version": "2.8.0", "language": "python", "license": "Apache-2.0", "repository_summary": "TensorFlow is an open source machine learning framework for everyone.", "repo_url": "https://github.com/tensorflow/tensorflow", "impact": {"cvss2_score": 3.3, "cvss2_severity": "LOW", "vector_string": "CVSS:3.1/AV:L/AC:H/PR:L/UI:N/S:U/C:H/I:H/A:N", "av": "LOCAL", "ac": "HIGH", "pr": "LOW", "c": "HIGH", "i": "HIGH", "a": "NONE", "impact_score": 5.2, "exploitability_score": 1.0, "cvss3_score": 6.3, "cvss3_severity": "MEDIUM", "ui": "NONE", "s": "UNCHANGED"}, "ochrona_severity_score": 6.3}